src/shared/loop-util.c

```diff
@@ -103,14 +110,12 @@ int loop_device_make_full(
                 loop = open(loopdev, O_CLOEXEC|O_NONBLOCK|O_NOCTTY|open_flags);
                 if (loop < 0)
                         return -errno;
-                if (ioctl(loop, LOOP_SET_FD, fd) < 0) {
-                        if (errno != EBUSY)
-                                return -errno;
-
-                        if (++n_attempts >= 64) /* Give up eventually */
-                                return -EBUSY;
-                } else
+                if (ioctl(loop, LOOP_SET_FD, fd) >= 0)
```

Comment:
so if this ioctl fails (which is likely when someone already uses the device, as we will get EBUSY then) then we'll end up calling LOOP_CLR_FD on the device, which will detach the fd of someone else.